src/Xamarin.Android.Build.Tasks/Tasks/GenerateJavaStubs.cs

```diff
@@ -296,8 +297,8 @@ Dictionary<string, ITaskItem> MaybeGetArchAssemblies (Dictionary<AndroidTargetAr
 
 		void GenerateAdditionalProviderSources (NativeCodeGenState codeGenState, IList<string> additionalProviders)
 		{
-			if (NativeAot) {
-				Log.LogDebugMessage ("Skipping MonoRuntimeProvider generation for NativeAot");
+			if (!string.Equals (AndroidRuntime, "MonoVM", StringComparison.OrdinalIgnoreCase)) {
```

Comment:
For forward compatibility, I would add an enum in XABT which contains the supported targets and a helper method in `MonoAndroidHelper` to parse strings and return the correct enum value.

Also, I think it would be a good idea to make `MonoVM`, `CoreCLR` and `NativeAOT` `const string` members in some class (`KnownRuntimes`?)

Together, the two changes would minimize potential for mistakes in code.

Reply:
I'll add a helper/enum for C#.